controllers/designate_controller.go

```diff
@@ -1059,9 +1059,55 @@ func (r *DesignateReconciler) generateServiceConfigMaps(
 	// - %-config configmap holding minimal designate config required to get the service up, user can add additional files to be added to the service
 	// - parameters which has passwords gets added from the ospSecret via the init container
 	//
+	Log := r.GetLogger(ctx)
 
 	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel(designate.ServiceName), map[string]string{})
+	replicas := int(*instance.Spec.DesignateBackendbind9.Replicas)
+	secretName := designate.DesignateBindKeySecret
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: instance.Namespace,
+		},
+	}
+
+	// Fetch the secret, create it if not found
+	_, err := controllerutil.CreateOrUpdate(ctx, h.GetClient(), secret, func() error {
+		secret.Labels = cmLabels
+		if secret.StringData == nil {
+			secret.StringData = make(map[string]string)
+		}
+
+		newKeysMap := make(map[string]string)
+
+		for i := 0; i < replicas; i++ {
+			keyName := fmt.Sprintf("rndc-key-%v", i)
```

Comment:
Can you put the fmt string in a constant in the designate pkg or similar? It's kind of a shared detail.

Reply:
Sure, sounds good to me, I will do that.